source/NVDAObjects/UIA/winConsoleUIA.py

```diff
@@ -430,3 +439,28 @@ def event_UIA_notification(self, **kwargs):
 		significantly improved and #11002 can be completely mitigated.
 		"""
 		log.debugWarning(f"Notification event blocked to avoid double-report: {kwargs}")
+
+
+class NotificationsBasedWinTerminalUIA(UIA):
+	"""
+	An overlay class for Windows Terminal (wt.exe) that uses UIA notification
+	events provided by the application to speak new text.
+	"""
+
+	#: Override the role, which is controlTypes.Role.STATICTEXT by default.
+	role = controlTypes.Role.TERMINAL
+	#: New line text is announced using UIA notification events
+	announceNewLineText = False
+
+	def event_UIA_notification(self, notificationKind, notificationProcessing, displayString, activityId):
```

Comment:
can this function header get typing information added?

Reply:
There is no typing on this method signature in the superclass either.

Reply:
Can you add types where they are known/obvious/simple?
This is still introducing new code without typing.